src/base/utils/fs.h

```diff
@@ -34,58 +35,38 @@
 
 #include <QString>
 
-namespace Utils::Fs
-{
-    /**
-     * Converts a path to a string suitable for display.
-     * This function makes sure the directory separator used is consistent
-     * with the OS being run.
-     */
-    QString toNativePath(const QString &path);
-
-    /**
-     * Converts a path to a string suitable for processing.
-     * This function makes sure the directory separator used is independent
-     * from the OS being run so it is the same on all supported platforms.
-     * Slash ('/') is used as "uniform" directory separator.
-     */
-    QString toUniformPath(const QString &path);
-
-    /**
-     * If `path is relative then resolves it against `basePath`, otherwise returns the `path` itself
-     */
-    QString resolvePath(const QString &relativePath, const QString &basePath);
+class QDateTime;
 
-    /**
-     * Returns the file extension part of a file name.
-     */
-    QString fileExtension(const QString &filename);
+class Path;
 
-    QString fileName(const QString &filePath);
-    QString folderName(const QString &filePath);
-    qint64 computePathSize(const QString &path);
-    bool sameFiles(const QString &path1, const QString &path2);
-    QString toValidFileSystemName(const QString &name, bool allowSeparators = false
-            , const QString &pad = QLatin1String(" "));
-    bool isValidFileSystemName(const QString &name, bool allowSeparators = false);
-    qint64 freeDiskSpaceOnPath(const QString &path);
-    QString branchPath(const QString &filePath, QString *removed = nullptr);
-    bool sameFileNames(const QString &first, const QString &second);
-    QString expandPath(const QString &path);
-    QString expandPathAbs(const QString &path);
-    bool isRegularFile(const QString &path);
-
-    bool smartRemoveEmptyFolderTree(const QString &path);
-    bool forceRemove(const QString &filePath);
-    void removeDirRecursive(const QString &path);
+namespace Utils::Fs
+{
+    qint64 computePathSize(const Path &path);
```

Comment:
It might be a good idea to sort this block alphabetically so that functions with similar prefixes will be placed next to each other.

Reply:
Slightly reordered declarations but still using logical relations.